core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -141,4 +138,34 @@ private static String stripTrailingSlash(String path) {
     }
     return result;
   }
+
+  private static String dataLocation(Map<String, String> properties, String tableLocation, boolean isObjectStore) {
+    String dataLocation = properties.get(TableProperties.WRITE_DATA_LOCATION);
+    if (dataLocation == null) {
+      String deprecatedProperty = null;
```

Comment:
I think we can use `String deprecatedProperty = isObjectStorage ? TableProperties.OBJECT_STORE_PATH : TableProperties.WRITE_FOLDER_STORAGE_LOCATION`, then `deprecatedProperty` is always not null, and you don't need to do null check for the warning, and we can avoid having that warning helper method.

Reply:
Made the change. Thanks for the suggestion.